jirazzz

```diff
@@ -276,8 +276,8 @@
                    {:json {:fields (merge
                                      (select-keys opts [:summary
                                                         :description])
-                                     {:project {:id (:project-id opts)}
-                                      :issuetype {:id (:type-id opts)}
+                                     {:project {:key (:project-id opts)}
+                                      :issuetype {:name (:type-id opts)}
```

Comment:
`type-id` is generated from `get-meta`:
https://github.com/rwstauner/jirazzz/blob/main/jirazzz#L209-L217

The `issue-type` configuration string is used to identify the correct issuetype from the meta response: https://github.com/rwstauner/jirazzz/blob/main/test/jirazzz/test_util.clj#L137-L147

If `type-id` was just the string from the config, it wouldn't be a map (so passing it to `:name` would return `nil`).

It's possible that at this point `type-id` was nil because the `issue-type` in the config didn't find a match in the project response.

Reply:
If for some reason `createmeta` isn't returning an issuetype that you want to use anyway, we could probably tweak the `get-meta` code to not overwrite `type-id` if `issue-type` wasn't provided.  That way a `type-id` in the config would pass through.

From what I know about Jira, however, trying to create an issue with a type that isn't included in the `createmeta` response will likely be rejected anyway.